.drone.starlark

```diff
@@ -0,0 +1,950 @@
+			'pull': 'always',
+		}]
+
+	return []
+
+def owncloudService(version, phpVersion, name = 'server', path = '/var/www/owncloud/server', ssl = True):
+	if ssl:
+		environment = {
+			'APACHE_WEBROOT': path,
+			'APACHE_CONFIG_TEMPLATE': 'ssl',
+			'APACHE_SSL_CERT_CN': 'server',
+			'APACHE_SSL_CERT': '/var/www/owncloud/%s.crt' % name,
+			'APACHE_SSL_KEY': '/var/www/owncloud/%s.key' % name
+		}
+	else:
+		environment = {
+			'APACHE_WEBROOT': path
+		}
+
+	return [{
+		'name': name,
+		'image': 'owncloudci/php:%s' % phpVersion,
+		'pull': 'always',
+		'environment': environment,
+		'command': [
+			'/usr/local/bin/apachectl',
+			'-e',
+			'debug',
+			'-D',
+			'FOREGROUND'
+		]
+	}]
+
+def installCore(version, db):
+	host = db.split(':')[0]
+	name = host
+
+	username = 'owncloud'
+	password = 'owncloud'
+	database = 'owncloud'
+
+	if name == 'mariadb':
+		name = 'mysql'
+
+	if name == 'postgres':
+		name = 'pgsql'
+
+	if name == 'oracle':
+		name = 'oci'
+		username = 'system'
+		password = 'oracle'
+		database = 'XE'
+
+	return {
+		'name': 'install-core',
+		'image': 'owncloudci/core',
+		'pull': 'always',
+		'settings': {
+			'version': version,
+			'core_path': '/var/www/owncloud/server',
+			'db_type': name,
+			'db_name': database,
+			'db_host': host,
+			'db_username': username,
+			'db_password': password
+		}
+	}
+
+def installTestrunner(phpVersion):
+	return {
+		'name': 'install-testrunner',
+		'image': 'owncloudci/php:%s' % phpVersion,
+		'pull': 'always',
+		'commands': [
+			'git clone -b master --depth=1 https://github.com/owncloud/core.git /var/www/owncloud/testrunner',
+			'cp -r /var/www/owncloud/server/apps/%s /var/www/owncloud/testrunner/apps/' % config['app'],
+			'cd /var/www/owncloud/testrunner',
+			'make install-composer-deps vendor-bin-deps'
+		]
+	}
+
+def installExtraApps(phpVersion, extraApps):
+	if extraApps == []:
+		return None
+
+	commandArray = []
+	for app in extraApps:
+		commandArray.append('git clone https://github.com/owncloud/%s.git /var/www/owncloud/server/apps/%s' % (app, app))
+		commandArray.append('cp -r /var/www/owncloud/server/apps/%s /var/www/owncloud/testrunner/apps/' % app)
+		commandArray.append('cd /var/www/owncloud/server')
+		commandArray.append('php occ a:l')
+		commandArray.append('php occ a:e %s' % app)
+		commandArray.append('php occ a:l')
+
+	return {
+		'name': 'install-extra-apps',
+		'image': 'owncloudci/php:%s' % phpVersion,
+		'pull': 'always',
+		'commands': commandArray
+	}
+
+def setupServerAndApp(phpVersion):
+	return {
+		'name': 'setup-server-%s' % config['app'],
+		'image': 'owncloudci/php:%s' % phpVersion,
+		'pull': 'always',
+		'commands': [
+			config['appInstallCommand'] if 'appInstallCommand' in config else ':',
+			'cd /var/www/owncloud/server',
+			'php occ a:l',
+			'php occ a:e %s' % config['app'],
+			'php occ a:e testing',
+			'php occ a:l',
+			'php occ config:system:set trusted_domains 1 --value=server',
+			'php occ log:manage --level 0',
```

Comment:
```suggestion
			'php occ log:manage --level 2',
```
or allow it to be set in the config

Reply:
done - log level defaults to 2 and can be overridden in the config